pkg/lib/packagekit.js

```diff
@@ -91,26 +91,33 @@
 }
 
 /**
- * Figure out whether PackageKit is available and usable
+ * Figure out whether PackageKit is available and usable, and return
+ * details of why not.
  */
-export function detect() {
+export function detect_with_details() {
     function dbus_detect() {
         return call("/org/freedesktop/PackageKit", "org.freedesktop.DBus.Properties",
                     "Get", ["org.freedesktop.PackageKit", "VersionMajor"])
-                .then(() => true,
-                      () => false);
+                .then(() => ({ available: true }),
+                      (error) => ({ available: false, reason: "not-found", error: error.toString() }));
     }
 
     return cockpit.spawn(["findmnt", "-T", "/usr", "-n", "-o", "VFS-OPTIONS"])
             .then(options => {
                 if (options.split(",").indexOf("ro") >= 0)
-                    return false;
+                    return { available: false, reason: "immutable-os" };
                 else
                     return dbus_detect();
             })
             .catch(dbus_detect);
 }
 
+/* Non-detailed versions of detect_with_details.
+ */
+export function detect() {
+    return detect_with_details().then(details => details.available);
+}
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18240-20230130-105647-26c41b66-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)